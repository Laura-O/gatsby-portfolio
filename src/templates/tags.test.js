import React from 'react';
import { shallow } from 'enzyme';

import Tags from './tags';
import {
  frontmatterPropType,
  sitePropType,
  imagePropType,
  markdownRemarkPropType,
} from '../../test/fixtures';

describe('blog', () => {
  test('1 tag', () => {
    const renderProps = {
      data: {
        allMarkdownRemark: {
          edges: [
            {
              node: {
                ...frontmatterPropType,
              },
            },
          ],
          totalCount: 1,
        },
      },
      pathContext: {
        tag: 'tag1',
      },
    };
    const wrapper = shallow(<Tags {...renderProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('2 tags', () => {
    const renderProps = {
      data: {
        allMarkdownRemark: {
          edges: [
            {
              node: {
                ...frontmatterPropType,
              },
            },
          ],
          totalCount: 2,
        },
      },
      pathContext: {
        tag: 'tag1',
      },
    };
    const wrapper = shallow(<Tags {...renderProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
