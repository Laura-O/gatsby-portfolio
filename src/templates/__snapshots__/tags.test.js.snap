// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blog 1 tag 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tags
    data={
      Object {
        "allMarkdownRemark": Object {
          "edges": Array [
            Object {
              "node": Object {
                "frontmatter": Object {
                  "date": "date",
                  "excerpt": "excerpt",
                  "image": Object {
                    "childImageSharp": Object {
                      "fluid": Object {
                        "sizes": "image",
                      },
                    },
                  },
                  "path": "path",
                  "tags": Array [
                    "tag1",
                    "tag2",
                  ],
                  "timeToRead": 10,
                  "title": "title",
                },
              },
            },
          ],
          "totalCount": 1,
        },
      }
    }
    pageContext={
      Object {
        "tag": "tag1",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          1 post tagged with "tag1"
        </h1>,
        <ul>
          <li>
            <GatsbyLink
              to="path"
            >
              title
            </GatsbyLink>
          </li>
        </ul>,
        <GatsbyLink
          to="/tags"
        >
          All tags
        </GatsbyLink>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "1 post tagged with \\"tag1\\"",
        },
        "ref": null,
        "rendered": "1 post tagged with \\"tag1\\"",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <GatsbyLink
                to="path"
              >
                title
              </GatsbyLink>
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "path",
            "nodeType": "host",
            "props": Object {
              "children": <GatsbyLink
                to="path"
              >
                title
              </GatsbyLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "title",
                "to": "path",
              },
              "ref": null,
              "rendered": "title",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "All tags",
          "to": "/tags",
        },
        "ref": null,
        "rendered": "All tags",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            1 post tagged with "tag1"
          </h1>,
          <ul>
            <li>
              <GatsbyLink
                to="path"
              >
                title
              </GatsbyLink>
            </li>
          </ul>,
          <GatsbyLink
            to="/tags"
          >
            All tags
          </GatsbyLink>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "1 post tagged with \\"tag1\\"",
          },
          "ref": null,
          "rendered": "1 post tagged with \\"tag1\\"",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <GatsbyLink
                  to="path"
                >
                  title
                </GatsbyLink>
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "path",
              "nodeType": "host",
              "props": Object {
                "children": <GatsbyLink
                  to="path"
                >
                  title
                </GatsbyLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "title",
                  "to": "path",
                },
                "ref": null,
                "rendered": "title",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "All tags",
            "to": "/tags",
          },
          "ref": null,
          "rendered": "All tags",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`blog 2 tags 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tags
    data={
      Object {
        "allMarkdownRemark": Object {
          "edges": Array [
            Object {
              "node": Object {
                "frontmatter": Object {
                  "date": "date",
                  "excerpt": "excerpt",
                  "image": Object {
                    "childImageSharp": Object {
                      "fluid": Object {
                        "sizes": "image",
                      },
                    },
                  },
                  "path": "path",
                  "tags": Array [
                    "tag1",
                    "tag2",
                  ],
                  "timeToRead": 10,
                  "title": "title",
                },
              },
            },
          ],
          "totalCount": 2,
        },
      }
    }
    pageContext={
      Object {
        "tag": "tag1",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          2 posts tagged with "tag1"
        </h1>,
        <ul>
          <li>
            <GatsbyLink
              to="path"
            >
              title
            </GatsbyLink>
          </li>
        </ul>,
        <GatsbyLink
          to="/tags"
        >
          All tags
        </GatsbyLink>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "2 posts tagged with \\"tag1\\"",
        },
        "ref": null,
        "rendered": "2 posts tagged with \\"tag1\\"",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <GatsbyLink
                to="path"
              >
                title
              </GatsbyLink>
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "path",
            "nodeType": "host",
            "props": Object {
              "children": <GatsbyLink
                to="path"
              >
                title
              </GatsbyLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "title",
                "to": "path",
              },
              "ref": null,
              "rendered": "title",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "All tags",
          "to": "/tags",
        },
        "ref": null,
        "rendered": "All tags",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            2 posts tagged with "tag1"
          </h1>,
          <ul>
            <li>
              <GatsbyLink
                to="path"
              >
                title
              </GatsbyLink>
            </li>
          </ul>,
          <GatsbyLink
            to="/tags"
          >
            All tags
          </GatsbyLink>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2 posts tagged with \\"tag1\\"",
          },
          "ref": null,
          "rendered": "2 posts tagged with \\"tag1\\"",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <GatsbyLink
                  to="path"
                >
                  title
                </GatsbyLink>
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "path",
              "nodeType": "host",
              "props": Object {
                "children": <GatsbyLink
                  to="path"
                >
                  title
                </GatsbyLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "title",
                  "to": "path",
                },
                "ref": null,
                "rendered": "title",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "All tags",
            "to": "/tags",
          },
          "ref": null,
          "rendered": "All tags",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
